src/backend/aspen/app/views/sample.py

```diff
@@ -447,6 +446,33 @@ def create_sample():
                     user.group.name,
                 ]
 
+            location_id = data["sample"].get("location_id", None)
+            valid_location: Optional[Location] = None
+            if location_id:
+                try:
+                    valid_location = (
+                        g.db_session.query(Location)
+                        .filter(Location.id == location_id)
+                        .one()
+                    )
+                except NoResultFound:
+                    sentry_sdk.capture_message(
+                        f"No valid location for id {location_id}"
+                    )
+                    raise ex.ServerException("Invalid location id for sample")
```

Comment:
Minor, but I think this actually would be a bad request, not the server's fault? I mean, it certainly _could_ be the server's fault if the previous API call to get location options went screwy, but this feels more like it should be "hey, you sent us a bad location_id, that's on you!" This feels like a 400 category, not a 500 category.